### PR TITLE
T-compiler meeting agenda: add Zulip discussion links to MCPs and FCPs

### PR DESCRIPTION
**Pull request by apiraino**
MCPs and FCPs all have a corresponding Zulip chat link for discussion before being approved (or declined).

In order to facilitate reaching this discussion happening on Zulip, this patch retrieves the first comment of an MCP/FCP, parses the corresponding Zulip link and add it to the T-compiler agenda.

Example of how the list of MCP/FCP look **before** this patch:
- Old MCPs (not seconded, take a look)
  - "Consistently use "region" terminology in later stages of the compiler" [compiler-team#634](https://github.com/rust-lang/compiler-team/issues/634) (last review activity: 3 months ago)

Example of how the list of MCP/FCP look **after** this patch:
- Old MCPs (not seconded, take a look)
  - "Consistently use "region" terminology in later stages of the compiler" [compiler-team#634](https://github.com/rust-lang/compiler-team/issues/634) ([Zulip](https://rust-lang.zulipchat.com/#narrow/stream/233931-xxx/topic/Consistently.20use.20.22region.22.20terminology.20in.20.E2.80.A6.20compiler-team.23634)) (last review activity: 3 months ago)

This idea was mentioned in a T-compiler meeting ([on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202023-06-29/near/370797526), of course).

Thanks for looking at this patch!

r? @spastorino

### CODE REVIEW

File: src/actions.rs [hunk 1]
```diff
@@ -124,8 +130,21 @@ impl<'a> Action for Step<'a> {
                     let query = query.clone();
                     handles.push(tokio::task::spawn(async move {
                         let _permit = semaphore.acquire().await?;
+                        let mcps_groups = [
+                            "mcp_new_not_seconded",
+                            "mcp_old_not_seconded",
+                            "mcp_accepted",
+                            "in_pre_fcp",
+                            "in_fcp",
+                        ];
```
Comment (apiraino) [inline src/actions.rs:139]: I am retrieving the Zulip links only for these FCPs/MCPs

File: src/actions.rs [hunk 2]
```diff
@@ -124,8 +130,21 @@ impl<'a> Action for Step<'a> {
...
                         let issues = query
-                            .query(&repository, name == "proposed_fcp", &gh)
+                            .query(
+                                &repository,
+                                name == "proposed_fcp",
+                                mcps_groups.contains(&name.as_str())
+                                    && repository.full_name.contains("rust-lang/compiler-team"),
```
Comment (apiraino) [inline src/actions.rs:145]: ... and that are on the `rust-lang/compiler-team` repository (easier way).

File: src/github.rs [hunk 3]
```diff
@@ -480,6 +480,18 @@ impl Issue {
...
+    pub async fn get_first_comment(&self, client: &GithubClient) -> anyhow::Result<Vec<Comment>> {
+        let comment_url = format!(
+            "{}/issues/{}/comments?page=1&per_page=1",
+            self.repository().url(),
+            self.number,
+        );
+        Ok(client
+            .json::<Vec<Comment>>(client.get(&comment_url))
+            .await?)
+    }
```
Comment (apiraino) [inline src/github.rs:493]: not too happy about this but I don't know another to ask GitHub to "return the first comment of an issue"

File: src/github.rs [hunk 4]
```diff
@@ -1588,12 +1601,13 @@ impl<'q> IssuesQuery for Query<'q> {
...
 
         let mut issues_decorator = Vec::new();
+        let re = regex::Regex::new("https://github.com/rust-lang/|/").unwrap();
+        let re_zulip_link = regex::Regex::new(r"\[stream\]:\s").unwrap();
         for issue in issues {
             let fcp_details = if include_fcp_details {
                 let repository_name = if let Some(repo) = issue.repository.get() {
                     repo.repository.clone()
                 } else {
-                    let re = regex::Regex::new("https://github.com/rust-lang/|/").unwrap();
```
Comment (apiraino) [inline src/github.rs:1596]: chore: this was moved outside the loop (tini tiny optimization...)

**Review by spastorino — APPROVED** [review]

**Pull request merged**